### DISCUSSION TITLE
[Unstable] Explosion will remove nearby explosives contained in detonators.

### DISCUSSION THREAD

**Discussion by zhu-rengong** [FakeFishGames/Barotrauma#14669] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

As the title said.

https://github.com/user-attachments/assets/8e7dd549-e2ed-4218-962a-092d6595e0cc

### Reproduction steps

1. spawn a compound N and place it inside any type of detonator
2. destroy the detonator by any means

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Single player

### Version

Other

### -

v1.6.9.0

### Which operating system did you encounter this bug on?

Windows

**Comment by zhu-rengong** [comment]
Maybe chain explosion is more reasonable more than removal

**Comment by zhu-rengong** [comment]
Not all methods can be used to reproduce, for example, if you open the health interface and then eat Flash Powder, it will not cause the compound N in the detonator to be removed. However, you can still reproduce it by this method:

https://github.com/user-attachments/assets/5f3c71bf-c7ea-40a8-be83-241f67e92d34

**Comment by Jasontti** [comment]
Tested the fix for this. No issues found.
Now detonators only take damage from their own explosion.
Closing discussion as completed.
